{"files":[{"patch":"@@ -2,1 +2,1 @@\n- * Copyright (c) 2021, 2023, Oracle and\/or its affiliates. All rights reserved.\n+ * Copyright (c) 2021, 2025, Oracle and\/or its affiliates. All rights reserved.\n@@ -109,1 +109,2 @@\n-                e.getCause().printStackTrace(System.out);\n+                System.err.println(\"test failure:\");\n+                e.getCause().printStackTrace();\n@@ -122,2 +123,1 @@\n-            String msg = e.getCause() == null ? e.getMessage() : e.getCause().getMessage();\n-            System.err.println(\"MSG RTE: \" + msg);\n+            final String msg = e.getCause() == null ? e.getMessage() : e.getCause().getMessage();\n@@ -125,3 +125,10 @@\n-            assertNotNull(msg);\n-            assertTrue(msg.contains(\"Network is unreachable\")\n-                        || msg.contains(\"No route to host\") || msg.contains(\"Connection timed out\"));\n+            if (msg != null &&\n+                    (msg.contains(\"Network is unreachable\")\n+                            || msg.contains(\"No route to host\")\n+                            || msg.contains(\"Connection timed out\"))) {\n+                \/\/ got the expected exception\n+                System.out.println(\"Received expected RuntimeException message: \" + msg);\n+            } else {\n+                \/\/ propagate the unexpected exception\n+                throw e;\n+            }\n@@ -129,3 +136,2 @@\n-            String msg = ex.getCause() == null ? ex.getMessage() : ex.getCause().getMessage();\n-            System.err.println(\"MSG: \" + msg);\n-            assertTrue(msg != null &&\n+            final String msg = ex.getCause() == null ? ex.getMessage() : ex.getCause().getMessage();\n+            if (msg != null &&\n@@ -135,1 +141,7 @@\n-                        || msg.contains(\"Timeout exceeded while waiting for a connection\")));\n+                        || msg.contains(\"Timeout exceeded while waiting for a connection\"))) {\n+                \/\/ got the expected exception\n+                System.out.println(\"Received expected NamingException message: \" + msg);\n+            } else {\n+                \/\/ propagate the unexpected exception\n+                throw ex;\n+            }\n","filename":"test\/jdk\/com\/sun\/jndi\/ldap\/LdapPoolTimeoutTest.java","additions":23,"deletions":11,"binary":false,"changes":34,"status":"modified"}]}